### ISSUE TITLE
the output of latent space of LSTM

### ISSUE THREAD

**Issue by YanhaoZhang** [qinnzou/Robust-Lane-Detection#22]
Hi, thanks a lot for publishing the code. It is a very interesting work. I am currently able to run the code with my own dataset. However, I notice that the latent space by LSTM is also an output for the training phase (`test`): https://github.com/qinnzou/Robust-Lane-Detection/blob/858954313edbc71ec937c96dc7b7368d08f59829/LaneDetectionCode/model.py#L66. It is also used to calculate the training loss as https://github.com/qinnzou/Robust-Lane-Detection/blob/858954313edbc71ec937c96dc7b7368d08f59829/LaneDetectionCode/train.py#L17. 

May I ask why the `test` is also considered when calculating the loss? Many thanks :)
